xclim/indices/_agro.py

```diff
@@ -944,6 +950,8 @@ def standardized_precipitation_index(
     ----------
     :cite:cts:`mckee_relationship_1993`
     """
+    uses_params = params is not None
+    uses_range = cal_range is not None
```

Comment:
Is there a possibility that a user could send in a variable that isn't "Truthy"? In all honesty, I kind of like this pattern, but you could probably get away with `if params:` below.

Reply:
Weird, `if params.any()` takes between 10 and 20 sec. It might be better if I stick to `if params is not None`

Reply:
I've removed `uses_range` for now, if you change idea on `uses_param` later just let me know!